### PR TITLE
kd_L2322_fert_usa

### PR DESCRIPTION
**Pull request by kdorheim**
#724

### CODE REVIEW

File: R/zchunk_L2322.Fert_USA.R [hunk 1]
```diff
@@ -52,168 +55,461 @@ module_gcam.usa_L2322.Fert_USA_DISABLED <- function(command, ...) {
...
+    L2322.SubsectorLogit_Fert %>%
+      filter(region == "USA", supplysector == aglu.FERT_NAME, subsector != "Imports") %>%
+      mutate(region = region) %>%
+      select(region, supplysector, subsector) ->
+      L2322.DeleteSubsector_USAFert
+
+    # Subset the supply sector keywords for fertilizer sector in the USA region.
+    L2322.FinalEnergyKeyword_Fert %>%
+      filter(region == "USA") %>%
+      mutate(`final.energy` = "none") ->
```
Comment (bpbond) [inline R/zchunk_L2322.Fert_USA.R]: Don't need backticks around `final.energy`

File: R/zchunk_L2322.Fert_USA.R [hunk 2]
```diff
@@ -52,168 +55,461 @@ module_gcam.usa_L2322.Fert_USA_DISABLED <- function(command, ...) {
...
+
+    # Use the subsector logit exponents of fertilizer sector to create
+    # a table of the subsector default technology share-weights for the US.
+    # that will be interpolated.
+    L2322.SubsectorLogit_USAFert %>%
+      select(region, supplysector, subsector) %>%
+      mutate(`apply.to` = "share-weight") %>%
+      mutate(`from.year` = max(BASE_YEARS)) %>%
+      mutate(`to.year` = max(MODEL_YEARS)) %>%
+      mutate(`interpolation.function` = "fixed") ->
```
Comment (bpbond) [inline R/zchunk_L2322.Fert_USA.R]: Same as above for these lines - no backticks needed.

**Comment by bpbond** [comment]
Holding on @mwisepnnl review.

**Comment by gokuliyer** [comment]
@bpbond: I just spoke with @mwisepnnl and we decided that I will do this before I leave tomorrow.

File: R/zchunk_L2322.Fert_USA.R [hunk 3]
```diff
@@ -1,46 +1,49 @@
...
-#' original data system was \code{L2322.Fert_USA.R} (gcam-usa level2).
-#' @details Describe in detail what this chunk does.
+#' the generated outputs: \code{L2322.DeleteSubsector_USAFert}, \code{L2322.FinalEnergyKeyword_USAFert}, \code{L2322.FinalEnergyKeyword_Fert_USA}, \code{L2322.StubTech_Fert_USA}, \code{L2322.SubsectorLogit_USAFert}, \code{L2322.SubsectorShrwtFllt_USAFert}, \code{L2322.TechShrwt_USAFert}, \code{L2322.Production_USAFert}, \code{L2322.TechCoef_USAFer}, \code{L2322.StubTechProd_Fert_USA}, \code{L2322.StubTechCoef_Fert_USA}, \code{L2322.StubTechMarket_Fert_USA}, \code{L2322.SubsectorLogit_Fert_USA}, \code{L2322.Supplysector_Fert_USA}, \code{L2322.SubsectorShrwtFllt_Fert_USA}, \code{L2322.SubsectorInterp_Fert_USA},\code{L2322.SubsectorInterp_USAFert}
+#'  The corresponding file in the original data system was \code{L2322.Fert_USA.R} (gcam-usa level2).
+#' @details This chunk produces the logit tables for the fertilizer sector in GCAM-USA
```
Comment (gokuliyer) [inline R/zchunk_L2322.Fert_USA.R]: This is not just the logit tables, but includes tables to create the fertilizer sector in GCAM-USA. Please update the comment to "This chink produces tables to create the N fertilizer sector in GCAM-USA"

File: R/zchunk_L2322.Fert_USA.R [hunk 4]
```diff
@@ -52,168 +55,462 @@ module_gcam.usa_L2322.Fert_USA_DISABLED <- function(command, ...) {
...
+
+
     # ===================================================
-    # TRANSLATED PROCESSING CODE GOES HERE...
-    #
-    # If you find a mistake/thing to update in the old code and
-    # fixing it will change the output data, causing the tests to fail,
-    # (i) open an issue on GitHub, (ii) consult with colleagues, and
-    # then (iii) code a fix:
+    # In the old data system this chunk processed the user defined outputs,
```
Comment (gokuliyer) [inline R/zchunk_L2322.Fert_USA.R]: Thanks for the detailed commenting! Very useful.

**Comment by bpbond** [comment]
@kdorheim I'll take care of Gokul's comment and merge this. Nice job.

**Pull request merged**